api/internal/repository/nft_token.go

```diff
@@ -151,6 +152,24 @@ func (n *nftTokenDbImpl) CreateNft(nftToken *NftTokenModel) (*NftTokenModel, err
 	return nftToken, nil
 }
 
+func (n *nftTokenDbImpl) UpdateNft(nftToken *NftTokenModel) (*NftTokenModel, error) {
+	nftToken.CollectionAddress = strings.ToLower(nftToken.CollectionAddress)
```

Comment:
seems like you may need a generic function to check and convert the addresses, may create an issue.

Reply:
Yes. Created an issue.